bin/update_item_type.rb

```diff
@@ -0,0 +1,71 @@
+require "cluster"
+require "services"
+
+# A (ideally) one-time solution to update mono_multi_serial values.
+# Usage:
+# bundle exec ruby bin/update_item_type.rb 0
+# ...
+# bundle exec ruby bin/update_item_type.rb f
+
+Services.mongo!
+
+# Divide the collection by the last char in the Cluster._id
+id_last_char = ARGV.shift
+unless (("a".."f").to_a + ("0".."9").to_a).include?(id_last_char)
+  raise "provide a char 0..f to divide clusters by"
+end
+
+log = File.open("/tmp/update_item_type_#{id_last_char}.log", "w")
+log.puts "Start time: #{Time.now}"
+puts "logging to #{log.path}"
+
+old_item_type_rx = /^(mono|multi|serial)$/
+old_to_new = {
+  "mono" => "spm",
+  "multi" => "mpm",
+  "serial" => "ser"
+}
+
+clusters_changed_count = 0
+holdings_changed_count = 0
+marker = Services.progress_tracker.new(1000)
+
+# For each cluster:
+begin
+  Cluster.each do |c|
+    unsaved_changes = false
+    current_cluster_id = c._id.to_s
+    next unless current_cluster_id.end_with?(id_last_char)
```

Comment:
In theory `_id` is indexed and you could partition into ranges, but they aren't evenly distributed (so you couldn't just go based on the first character of `_id`). This idea should work well enough, even if it isn't likely to be the fastest.